client/components/chart/utils.js

```diff
@@ -207,71 +209,113 @@ export const getLine = ( xLineScale, yScale ) =>
 		.y( d => yScale( d.value ) );
 
 /**
- * Describes getXTicks
- * @param {array} uniqueDates - all the unique dates from the input data for the chart
+ * Calculate the maximum number of ticks allowed in the x-axis based on the width and layout of the chart
  * @param {integer} width - calculated page width
  * @param {string} layout - standard, comparison or compact chart types
  * @returns {integer} number of x-axis ticks based on width and chart layout
  */
-export const getXTicks = ( uniqueDates, width, layout ) => {
-	// caluclate the maximum number of ticks allowed in the x-axis based on the width
-	// and layout of the chart
-	let ticks = 16;
+const calculateMaxXTicks = ( width, layout ) => {
 	if ( width < 783 ) {
-		ticks = 7;
+		return 7;
 	} else if ( width >= 783 && width < 1129 ) {
-		ticks = 12;
+		return 12;
 	} else if ( width >= 1130 && width < 1365 ) {
 		if ( layout === 'standard' ) {
-			ticks = 16;
+			return 16;
 		} else if ( layout === 'comparison' ) {
-			ticks = 12;
+			return 12;
 		} else if ( layout === 'compact' ) {
-			ticks = 7;
+			return 7;
 		}
 	} else if ( width >= 1365 ) {
 		if ( layout === 'standard' ) {
-			ticks = 31;
+			return 31;
 		} else if ( layout === 'comparison' ) {
-			ticks = 16;
+			return 16;
 		} else if ( layout === 'compact' ) {
-			ticks = 12;
+			return 12;
 		}
 	}
-	if ( uniqueDates.length <= ticks ) {
-		return uniqueDates;
+
+	return 16;
+};
+
+/**
+ * Filter out irrelevant dates from uniqueDates if there are too many
+ * @param {array} uniqueDates - all the unique dates from the input data for the chart
+ * @returns {integer} Unique dates filtered
+ */
+const filterXAxisDates = uniqueDates => {
```

Comment:
Just so I understand what's going on here- does this function filter out dates by unique months?

Reply:
Right, it only takes the first date of each month. I tried to improve the function naming and JSDoc comments to make it clearer.